lib/chai/core/assertions.js

```diff
@@ -1285,130 +1286,104 @@ module.exports = function (chai, _) {
    * @name throw
    * @alias throws
    * @alias Throw
-   * @param {ErrorConstructor} constructor
-   * @param {String|RegExp} expected error message
+   * @param {Error|ErrorConstructor} errorLike
+   * @param {String|RegExp} errMatcher error message
    * @param {String} message _optional_
    * @see https://developer.mozilla.org/en/JavaScript/Reference/Global_Objects/Error#Error_types
    * @returns error for chaining (null if no error)
    * @namespace BDD
    * @api public
    */
 
-  function assertThrows (constructor, errMsg, msg) {
+  function assertThrows (errorLike, errMatcher, msg) {
     if (msg) flag(this, 'message', msg);
     var obj = flag(this, 'object');
+    var negate = flag(this, 'negate');
     new Assertion(obj, msg).is.a('function');
 
-    var thrown = false
-      , desiredError = null
-      , name = null
-      , thrownError = null;
-
-    if (arguments.length === 0) {
-      errMsg = null;
-      constructor = null;
-    } else if (constructor && (constructor instanceof RegExp || 'string' === typeof constructor)) {
-      errMsg = constructor;
-      constructor = null;
-    } else if (constructor && constructor instanceof Error) {
-      desiredError = constructor;
-      constructor = null;
-      errMsg = null;
-    } else if (typeof constructor === 'function') {
-      name = constructor.prototype.name;
-      if (!name || (name === 'Error' && constructor !== Error)) {
-        name = constructor.name || (new constructor()).name;
-      }
-    } else {
-      constructor = null;
+    if (errorLike instanceof RegExp || typeof errorLike === 'string' && arguments.length < 2) {
+      errMatcher = errorLike;
+      errorLike = null;
     }
```

Comment:
`arguments.length < 2` is really not needed. You're right here.
I think I've got confused because it's possible to throw a `String` instead of an `Error` object, but then we should do message checks instead, since there's no way a thrown `String` will also have a message, it will be the message itself.